### ISSUE TITLE
Script generates invalid play date data for an unplayed track

### ISSUE THREAD

**Issue by fvdpol** [fvdpol/MM-ExportITunesLibraryXml#13]
Reported on Media Monkey forum by  Teknojnky on Mon Oct 22, 2018 4:53 am

link: http://www.mediamonkey.com/forum/viewtopic.php?f=2&t=31680&sid=4c4111f5c9aba2c4fc277304e5064733&start=105#p450930

this script generates invalid play date data for an unplayed track.

here is a real itunes xml output, there is no key for play date.

```
		<key>7035</key>
		<dict>
			<key>Track ID</key><integer>7035</integer>
			<key>Size</key><integer>2748416</integer>
			<key>Total Time</key><integer>168803</integer>
			<key>Track Number</key><integer>13</integer>
			<key>Track Count</key><integer>17</integer>
			<key>Year</key><integer>1990</integer>
			<key>BPM</key><integer>134</integer>
			<key>Date Modified</key><date>2009-01-14T09:44:52Z</date>
			<key>Date Added</key><date>2018-10-22T02:16:03Z</date>
			<key>Bit Rate</key><integer>128</integer>
			<key>Sample Rate</key><integer>44100</integer>
			<key>Artwork Count</key><integer>1</integer>
			<key>Persistent ID</key><string>DD6E628FAB056EA5</string>
			<key>Track Type</key><string>File</string>
			<key>File Folder Count</key><integer>-1</integer>
			<key>Library Folder Count</key><integer>-1</integer>
			<key>Name</key><string>21st Century (Digital Boy)</string>
			<key>Artist</key><string>Bad Religion</string>
			<key>Album Artist</key><string>Bad Religion</string>
			<key>Album</key><string>Against the Grain</string>
			<key>Genre</key><string>Punk</string>
			<key>Kind</key><string>MPEG audio file</string>
			<key>Sort Artist</key><string>Bad Religion</string>
			<key>Location</key><string>file://localhost//sauron/media/audio/MP3/B/Bad%20Religion/1990%20-%20Against%20the%20Grain%20%5BMP3%5D/13%20-%20Bad%20Religion%20-%2021st%20Century%20(Digital%20Boy).mp3</string>
		</dict>

```

here is the script generated from mediamonkey for the same unplayed track (added dates will be different), with an invalid play date that is obviously in the past.

I confirmed in mediamonkey that there is no play date, and no play counts.

```
        <key>39283</key>
        <dict>   
            <key>Track ID</key><integer>39283</integer>
            <key>Name</key><string>21st Century (Digital Boy)</string>
            <key>Artist</key><string>Bad Religion</string>
            <key>Album Artist</key><string>Bad Religion</string>
            <key>Album</key><string>Against the Grain</string>
            <key>Kind</key><string>MPEG audio file</string>
            <key>Size</key><integer>2748416</integer>
            <key>Genre</key><string>Punk</string>
            <key>Total Time</key><integer>168803</integer>
            <key>Track Number</key><integer>13</integer>
            <key>Rating</key><integer>50</integer>
            <key>Year</key><integer>1990</integer>
            <key>Date Modified</key><date>2009-01-14T03:44:52</date>
            <key>Date Added</key><date>2007-12-28T16:51:22</date>
 *           <key>Play Date UTC</key><date>1899-12-30T00:00:00</date>
            <key>Bit Rate</key><integer>128</integer>
            <key>Sample Rate</key><integer>44100</integer>
            <key>Track Type</key><string>File</string>
            <key>File Folder Count</key><integer>-1</integer>
            <key>Library Folder Count</key><integer>-1</integer>
            <key>BPM</key><string>134</string>
            <key>Location</key><string>file://localhost//sauron/media/audio/MP3/B/Bad%20Religion/1990%20-%20Against%20the%20Grain%20%5BMP3%5D/13%20-%20Bad%20Religion%20-%2021st%20Century%20(Digital%20Boy).mp3</string>
        </dict>

```

I raise this issue, due to the import to plex, it imports this invalid play date of the past, into the FUTURE in plex's database.

So there is actually 2 problems, 1 with the export script, and 1 with the plex import.

https://forums.plex.tv/t/plex-music-rec ... /319231/15

**Comment by fvdpol** [comment]
Validated against the current development branch;

File #1;  played in MediaMonkey
![image](https://user-images.githubusercontent.com/1499454/48628739-3d823d80-e9b8-11e8-90e4-eb4cbbb4a384.png)

>> in exported xml:
![image](https://user-images.githubusercontent.com/1499454/48628890-8df99b00-e9b8-11e8-9f39-f1cc57655c48.png)


File #2; not played in MediaMonkey
![image](https://user-images.githubusercontent.com/1499454/48628792-568aee80-e9b8-11e8-82bf-c44dab116e70.png)

>> in exported xml:
![image](https://user-images.githubusercontent.com/1499454/48628919-9fdb3e00-e9b8-11e8-9309-10255523bda8.png)



Check behaviour if iTunes (12.9.0 windows):

File with a playcount/date:
![image](https://user-images.githubusercontent.com/1499454/48629153-33147380-e9b9-11e8-8e2b-5553aa4fdc5d.png)

>> note: MediaMonkey does not have the Play Date in unix time format --- TODO



File without a playcount/date:
![image](https://user-images.githubusercontent.com/1499454/48629398-c5b51280-e9b9-11e8-90e5-05c0d46cd1d2.png)


Conclusion: MediaMonkey (current dev branch) output is consistent with Apple iTunes for the Play Count / Date, except for missing date in unix timestamp format

**Comment by fvdpol** [comment]
Closing this issue; changes included in released version v1.6.3
(https://github.com/fvdpol/MM-ExportITunesLibraryXml/releases/tag/v1.6.3)
